hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/simple/HoodieSimpleIndex.java

```diff
@@ -144,7 +144,7 @@ protected HoodiePairData<HoodieKey, HoodieRecordLocation> fetchRecordLocationsFo
       HoodieData<HoodieKey> hoodieKeys, HoodieEngineContext context, HoodieTable hoodieTable,
       int parallelism) {
     List<String> affectedPartitionPathList =
-        hoodieKeys.map(HoodieKey::getPartitionPath).distinct().collectAsList();
+        hoodieKeys.map(HoodieKey::getPartitionPath).distinct(hoodieKeys.deduceNumPartitions()).collectAsList();
```

Comment:
why we cannot use the parallelism value received as input in the `distinct()` i.e. 
`hoodieKeys.map(HoodieKey::getPartitionPath).distinct(parallelism).collectAsList();`

Reply:
the case you had face, it will case parallelism too bigger if we use spark own default inference logic
![image](https://github.com/apache/hudi/assets/20125927/f47ab906-349d-4e15-81f7-a0cba52e81bd)